app/models/user.rb

```diff
@@ -88,9 +83,24 @@ def update_level
     update(level: user_level)
   end
 
-  def add_level_label(key, value = 'verified')
-    labels.find_or_create_by(key: key, scope: 'private')
-          .update!(value: value)
+  def update_state
+    @resulting_state = 'pending'
+
+    unless active?
+      if BarongConfig.list['activation_requirements'] <= (labels.with_private_scope.map { |l| { l.key => l.value }}.inject(:merge) || {})
+        @resulting_state = 'active'
+      end
+    end
+
```

Comment:
#FIXME BarongConfig should be a feature of Barong::App
A good name for Barong::App can be Barong::Config and we can add database support or vault based configs.